src/humanpy/hum_kin2.py

```diff
@@ -0,0 +1,298 @@
+            person_position_transform = numpy.zeros((4, 4))
+            person_position_transform[0:3,0:3] = person_position_transform_rot
+            person_position_transform[0:4,3] = person_transform[0:4,3]
+            person_position_transform[2,3] = 0.85
+            self.body.SetTransform(person_position_transform)
+        
+        ##Left arm
+        ul_angles = self.getUpperLimbAngles(tf, 'L')
+        if ul_angles is not None:
+            self.checkLimits(self.body.GetJoint('JLShoulder'), ul_angles, 0)
+            self.checkLimits(self.body.GetJoint('JLShoulder2'), ul_angles, 1)
+            self.checkLimits(self.body.GetJoint('JLShoulder3'), ul_angles, 2)
+            self.checkLimits(self.body.GetJoint('JLForearm'), ul_angles, 3)
+
+        #Right arm
+        ul_angles = self.getUpperLimbAngles(tf, 'R')
+        if ul_angles is not None: 
+            self.checkLimits(self.body.GetJoint('JRShoulder'), ul_angles, 0)
+            self.checkLimits(self.body.GetJoint('JRShoulder2'), ul_angles, 1)
+            self.checkLimits(self.body.GetJoint('JRShoulder3'), ul_angles, 2)        
+            self.checkLimits(self.body.GetJoint('JRForearm' ), ul_angles, 3)
+        
+        if self.enable_legs == True:        
+            #Left leg
+            ul_angles = self.getLowerLimbAngles(tf, 'L')
+            if ul_angles is not None:
+                self.checkLimits(self.body.GetJoint('JLThigh'), ul_angles, 0)
+                self.checkLimits(self.body.GetJoint('JLCalf'), ul_angles, 1)
+
+            #Right leg
+            ul_angles = self.getLowerLimbAngles(tf, 'R')
+            if ul_angles is not None:
+                self.checkLimits(self.body.GetJoint('JRThigh'), ul_angles, 0)
+                self.checkLimits(self.body.GetJoint('JRCalf'), ul_angles, 1)
+        
+        #TODO; head, neck  
+
+
+def humanInList(human, ids):
+    for id in ids:
+        if human.id == 'user_' + id: return True
+    return False
+        
+def addRemoveHumans(tf, humans, env, enable_legs=True, segway_sim=True):
+    import re
+    matcher = re.compile('.*user_(\\d+).*')    
+    all_tfs = tf.getFrameStrings()
+    all_human_ids = []
+    for frame_name in all_tfs:
+        match = matcher.match(frame_name)
+        if match is not None:
+            all_human_ids.append(match.groups()[0])
+
+    # removing
+    humans_to_remove = [x for x in humans if not humanInList(x, all_human_ids)]
+    for human in humans_to_remove:
+        env.RemoveKinBody(human);
+        human.destroy()
+        humans.remove(human)
+        
+    # adding
+    for id in all_human_ids:
+        found = False
+        for human in humans:
+            if human.id == 'user_' + id:
+                found = True
+                break
+        if not found:
+            humans.append(Orhuman('user_' + id, env, 
+                                  enable_legs=enable_legs,
+                                  segway_sim=segway_sim))
+
```

Comment:
Some functions seem duplicated from hum_kin1 and hum_kin2. Is that true and if so why?